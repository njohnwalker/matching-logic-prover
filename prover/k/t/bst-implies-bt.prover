claim \implies( \and( bst(variable("H"), variable("X"), variable("F"), variable("MIN"), variable("MAX"), .Patterns)
                    , .Patterns
                    )
              , \and( bt( variable("H"), variable("X"), variable("F"), .Patterns), .Patterns )
              )
strategy kt ; ( ( right-unfold-Nth(0,0) ; simplify ; direct-proof )
              | ( right-unfold-Nth(0,1) ; right-unfold-Nth(0,0) ; right-unfold-Nth(0,0) ; simplify ; direct-proof )
              | direct-proof
              | ( right-unfold-Nth(0,1) ) ; simplify ; direct-proof
              )
