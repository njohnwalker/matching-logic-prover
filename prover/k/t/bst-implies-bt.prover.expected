<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( bst ( variable ( "H" ) , variable ( "X" ) , variable ( "F" ) , variable ( "MIN" ) , variable ( "MAX" ) , .Patterns ) , .Patterns ) , \and ( bt ( variable ( "H" ) , variable ( "X" ) , variable ( "F" ) , .Patterns ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
      .
    </trace>
  </goal>
</prover>
